library/src/com/google/maps/android/geoJsonLayer/GeoJsonRenderer.java

```diff
@@ -0,0 +1,308 @@
+package com.google.maps.android.geoJsonLayer;
+
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.android.gms.maps.model.Polygon;
+import com.google.android.gms.maps.model.PolygonOptions;
+import com.google.android.gms.maps.model.Polyline;
+import com.google.android.gms.maps.model.PolylineOptions;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Observable;
+import java.util.Observer;
+
+/**
+ * Renders GeoJsonFeature objects onto the GoogleMap as Marker, Polyline and Polygon objects. Also
+ * removes GeoJsonFeature objects and redraws features when updated.
+ */
+class GeoJsonRenderer implements Observer {
+
+    private final static int POLYGON_OUTER_COORDINATE_INDEX = 0;
+
+    private final static int POLYGON_INNER_COORDINATE_INDEX = 1;
+
+    private final static Object FEATURE_NOT_ON_MAP_VALUE = null;
+
+    private final HashMap<GeoJsonFeature, Object> mFeatures;
+
+    private GoogleMap mMap;
+
+    /**
+     * Creates a new GeoJsonRender object
+     *
+     * @param features hashmap of features to store
+     * @param map      map to place GeoJsonFeature objects on
+     */
+    public GeoJsonRenderer(HashMap<GeoJsonFeature, Object> features, GoogleMap map) {
```

Comment:
make all methods in this class package-private as well (except for those for which this is not possible like those that override Observer methods)